src/config.rs

```diff
@@ -393,23 +393,103 @@ impl Config {
             Ok(())
         }
 
-        const ID_HEADER_PATTERN: &str = r"(?m)^\s*id\s*:[\sA-Za-z\d]*$";
+        /// returns true when the next occurrence of id header needs to be ignored
+        /// for cases when the section-line refers to a component definition
+        /// or if the section is a youtube type component
+        /// since user can put multiple instances of this component
+        /// in various places in the package referring to the same youtube video
+        /// having same id so ignoring these ids.
+        ///
+        /// in any other case returns false
+        fn ignore_next_id(section_line: &str) -> bool {
+            // Strip out initial '-- ' or '--- '
+            let section_line = section_line.trim_start_matches('-').trim_start();
+
+            let before_caption = section_line.split_once(':').map(|s| s.0);
+            if let Some(section) = before_caption {
+                let mut parts = section.splitn(2, ' ');
+
+                // in case of component definition, section-kind will be mandatory
+                // -- <optional: section-kind> section-name: <section-caption>
+                return match (parts.next(), parts.next()) {
+                    (Some(_kind), Some(_name)) => true,
+                    (Some(_name), None) if _name.contains("youtube") => true,
```

Comment:
`fpm::constants`, `YOUTUBE = "youtube"`

Reply:
this is not required anymore since `ft.youtube` doesn't use `id` argument for videos instead uses `v` argument now.